### ISSUE TITLE
`supporting_services_managed_in_dynatrace` setting to `true` still wipes out supporting service configuration

### ISSUE THREAD

**Issue by jeevanions** [dynatrace-oss/terraform-provider-dynatrace#297]
We are creating a connection for Azure subscription using `dynatrace_azure_credentials` Terraform resource. Setting the flag `supporting_services_managed_in_dynatrace` to `true` still removes any previously configured supporting services in Dynatrace Web UI. 

```
resource "dynatrace_azure_credentials" dynatrace_azure_creds" {
  active                       = true
  app_id                       = var.dynatrace_azure_cred_appid
  auto_tagging                 = true
  directory_id                 = data.azurerm_subscription.current.tenant_id
  label                        = "${var.account}"
  key                          = var.dynatrace_azure_cred_secret
  monitor_only_tagged_entities = false
  supporting_services_managed_in_dynatrace = true
}
```

**Comment by jeevanions** [comment]
Terraform version used 1.35.2 release today. I did try a very old version 1.26.0 and the issue is there in the old version.

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hello @jeevanions,

That behavior is pretty surprising. When implementing the resource `dynatrace_azure_service` I noticed that the REST endpoint `/azure/credentials/{id}` is meanwhile completely ignoring the supported services (contradicting what the documentation of the endpoint says).
As a matter of fact, we've implemented the resource `dynatrace_azure_service` *because* they couldn't get managed via `dynatrace_azure_credentials` anymore.

I also ran a couple of local tests, which confirmed the findings from back then.

Can you do me the favor and execute with these environment variables?
```
DYNATRACE_DEBUG=true
DYNATRACE_LOG_HTTP=terraform-provider-dynatrace.http.log
DYNATRACE_HTTP_RESPONSE=true
```

This should produce the file `terraform-provider-dynatrace.http.log` (and potentially `terraform-provider-dynatrace.http.err.log`) containing detailed logs of the HTTP traffic. Feel free to send the log files via email to terraform@dynatrace.com to avoid leaking any confidential data about your infrastructure here on GitHub.

Another question: Are you working with a SaaS environment of Dynatrace or do you have a Managed Dynatrace Cluster? If yes, what version is the cluster at?

**Comment by jeevanions** [comment]
@Dynatrace-Reinhard-Pilz  Apologies for the delayed response. 

This is so strange and confusing.

Step 1: I have created a separate Terraform script which simply has the `dynatrace_azure_credentials` with exactly the same environment as our original Terraform scripts. This time it worked fine and the `terraform-provider-dynatrace.http.log` logs are generated as expected. The monitoring services configuration is also retained.  Terraform version used here is Terraform v1.2.9

Step 2:  So I went back to our script (note it have more providers and complex). With all the suggested environment variable, the monitoring services configured in the Dynatrace web is wiped out and no logs generated. This is very consistent.  So not sure what I am doing wrong here. Below is the list of providers and versions we use. 

```
Terraform v1.2.9
on linux_amd64
+ provider registry.terraform.io/dynatrace-oss/dynatrace v1.26.0
+ provider registry.terraform.io/hashicorp/azuread v2.27.0
+ provider registry.terraform.io/hashicorp/azurerm v3.48.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.14.0
+ provider registry.terraform.io/hashicorp/local v2.4.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/template v2.2.0
+ provider registry.terraform.io/hashicorp/time v0.9.1
+ provider registry.terraform.io/hashicorp/tls v4.0.4
```

I have enabled `TF_LOG=Trace` but that has lot of noise from other providers. So not sure the best way to get the logs . Any Suggestions?

**Comment by jeevanions** [comment]
@Dynatrace-Reinhard-Pilz Finally found the issue. 

The setup we have here is generating Azure AD SP Secret on a time rotation. We have set this to be 2 mins for testing. Every-time the Terraform is run the SP Secret is updated which is consumed by the `dynatrace_azure_credentials` resource. Since there is a change it tries to update the azure credentials where it also wipes out the supporting services configuration managed in Dynatrace Web UI.. I will send you a sample code to the support email for your testing.

**Comment by jeevanions** [comment]
@Dynatrace-Reinhard-Pilz Did you get a chance to look at this?

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Sorry for the late response @jeevanions.
Would you be willing to jump on a working session together with me sometime next week? The output within the generated log files when you execute your config would be important to get evaluated. Second, during such a session we could run a special build of the provider that reveals even more information.

**Comment by jeevanions** [comment]
Are you available tomorrow? to have this call. 

But this issue can be easily replicated by changing the `key` arguments.

**Comment by Dynatrace-Reinhard-Pilz** [comment]
I've tried to replicate locally and I'm not getting the same results.
Would Thursday also be an option for you? You may choose a 30 minute time slot that works best for you. I'm located in the [CEST](https://www.timeanddate.com/time/zones/cest) time zone.

**Comment by jeevanions** [comment]
@Dynatrace-Reinhard-Pilz  Could you confirm your email address and we will drop in the zoom link?

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hello @jeevanions,
The upcoming version of the provider will support the use case we looked at during our session. Before an update the current status will get pulled for the supported services. 
You may also want to look at the resource [dynatrace_azure_service](https://registry.terraform.io/providers/dynatrace-oss/dynatrace/latest/docs/resources/azure_service) for alternative configuration. That resource got implemented because of limitations that got introduced to the original endpoint.

**Comment by jeevanions** [comment]
@Dynatrace-Reinhard-Pilz Thank you for a quick turnaround. Yeah we have looked at the `dynatrace_azure_service` but our use case case is slightly different. We are a centralised team provides a landing zone setup and leave the supporting services configuration to the respective team. We deal with multiple such team and it is hard for use to fullfill requirement for all these teams. So we decided to leave this config to respective team and managed from web ui.

**Comment by jeevanions** [comment]
> 

@Dynatrace-Reinhard-Pilz I hope this function "an update the current status will get pulled for the supported services" does not apply to the user opt out from `supporting_services_managed_in_dynatrace` feature.

**Comment by Dynatrace-Reinhard-Pilz** [comment]
You're correct

**Comment by nagachakithmck** [comment]
@Dynatrace-Reinhard-Pilz thanks for the update. Do you know which version this feature will be available ?
is it 1.39.0 ?

**Comment by kishikawa12** [comment]
Hi @nagachakithmck,

The fix for this particular issue was introduced in v1.37.0, but the latest release currently available is v1.39.0.
